File: app/src/main/java/com/norvera/confession/data/dao/GuideDao.java
package com.norvera.confession.data.dao;

import com.norvera.confession.data.models.GuideEntry;

import java.util.List;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Query;

@Dao
public abstract class GuideDao {

    @Query("SELECT * FROM  GUIDE_MAIN WHERE h_id = :guideId")
    public abstract LiveData<List<GuideEntry>> allGuidesForId(int guideId);

}

File: app/src/main/java/com/norvera/confession/data/dao/GuideDao.kt
package com.norvera.confession.data.dao

import com.norvera.confession.data.models.GuideEntry

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Query

@Dao
abstract class GuideDao {

    @Query("SELECT * FROM  GUIDE_MAIN WHERE h_id = :guideId")
    abstract fun allGuidesForId(guideId: Int): LiveData<List<GuideEntry>>

}
